plugins/cryptolib/cryptolib.cpp

```diff
@@ -411,6 +411,52 @@ void verifyPKIAlgorithm(const char * pkAlgorithm)
 }
 
 
+//---------------------------------------------------------
+// Helper that reads the given logical file into a string buffer
+// Throws if user does not have read permission or if FNF
+//---------------------------------------------------------
+void loadLFS(const char * lfs, IUserDescriptor * user, StringBuffer &sb)
+{
+    CDfsLogicalFileName lfn;
+    lfn.set(lfs);
+    try
+    {
+        Owned<IDistributedFile> df = queryDistributedFileDirectory().lookup(lfn, user, false, false, false, nullptr, defaultPrivilegedUser);//scope checks
+        if (!df)
+        {
+            throw MakeStringException(-1, "File Not Found");
+        }
+
+        unsigned numParts = df->numParts();
+        if (numParts != 1)
+        {
+            throw MakeStringException(-1, "Part count (%d) is incorrect", numParts);
+        }
+
+        IDistributedFilePart *part = df->getPart(0);
+
+        RemoteFilename rfn;
+        Owned<IFile> file = createIFile(part->getFilename(rfn));
+        if (!file->exists())
+        {
+            throw MakeStringException(-1, "File Not Found");
+        }
+
+        Owned<IFileIO> io = file->open(IFOread);
+        offset_t len = file->size();
+        char * buff = sb.reserve(len);
```

Comment:
trivial: slightly better to use reserveTruncate to avoid over allocation.

Reply:
Done. Must have missed that feature in the JLib docs :)